app/src/main/java/org/thoughtcrime/securesms/onboarding/pickname/PickDisplayNameViewModel.kt

```diff
@@ -0,0 +1,116 @@
+package org.thoughtcrime.securesms.onboarding.pickname
+
+import android.content.Context
+import androidx.annotation.StringRes
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import network.loki.messenger.R
+import org.session.libsession.snode.SnodeModule
+import org.session.libsession.utilities.SSKEnvironment.ProfileManagerProtocol.Companion.NAME_PADDED_LENGTH
+import org.session.libsession.utilities.TextSecurePreferences
+import org.session.libsignal.database.LokiAPIDatabaseProtocol
+import org.session.libsignal.utilities.KeyHelper
+import org.session.libsignal.utilities.hexEncodedPublicKey
+import org.thoughtcrime.securesms.crypto.KeyPairUtilities
+import org.thoughtcrime.securesms.dependencies.ConfigFactory
+
+class PickDisplayNameViewModel(
+    pickNewName: Boolean,
+    private val prefs: TextSecurePreferences,
+    private val configFactory: ConfigFactory
+): ViewModel() {
+    private val state = MutableStateFlow(if (pickNewName) pickNewNameState() else State())
+    val stateFlow = state.asStateFlow()
+
+    private val event = Channel<Event>()
+    val eventFlow = event.receiveAsFlow()
+
+    private val database: LokiAPIDatabaseProtocol
+        get() = SnodeModule.shared.storage
+
+    fun onContinue(context: Context) {
+        state.update { it.copy(displayName = it.displayName.trim()) }
+
+        val displayName = state.value.displayName
+
+        val keyPairGenerationResult = KeyPairUtilities.generate()
```

Comment:
Do we need those 4 lines here? Or should they be moved inside the else in the when below if they are only used there?

Reply:
@bemusementpark

Reply:
done